client/src/lib.rs

```diff
@@ -550,14 +551,17 @@ impl<V: vdaf::Client<16>> Client<V> {
         ))
     }
 
-    /// Upload a [`Report`] to the leader, per the [DAP specification][1]. The provided measurement
-    /// is sharded into two shares and then uploaded to the leader.
+    /// Upload one or more measurements to the leader, per the [DAP specification][1]. The provided
+    /// measurements are sharded into two shares and then uploaded to the leader.
     ///
     /// [1]: https://www.ietf.org/archive/id/draft-ietf-ppm-dap-07.html#name-uploading-reports
-    #[tracing::instrument(skip(measurement), err)]
-    pub async fn upload(&self, measurement: &V::Measurement) -> Result<(), Error> {
-        self.upload_with_time(measurement, Clock::now(&RealClock::default()))
-            .await
+    #[tracing::instrument(skip(measurements), err)]
+    pub async fn upload(&self, measurements: &[V::Measurement]) -> Result<(), Error> {
```

Comment:
I think it would be preferable if `Client::upload()` still handled just one report, and we added another method for sharding and submitting multiple reports. (or just relied on `upload_with_time()` as the most general API) Since many measurement types are already vectors, this would make the common case easier to understand. Besides, I think the primary use case for submitting multiple reports per request will be intermediaries collecting reports from multiple systems. Sharding and submitting multiple reports on the same system seems like it will mainly be needed for testing purposes and not practical applications.

Reply:
I totally agree -- I inherited this from Tim's original PR, and thinking about it this way, well, I reverted in 701c9239.